data/blog/publishing-binaries-on-npm.md

```diff
@@ -0,0 +1,376 @@
+     }
+
+     // Windows binaries end with .exe so we need to special case them.
+     const binaryName = process.platform === 'win32' ? 'my-binary.exe' : 'my-binary'
+
+     // Determine package name for this platform
+     const platformSpecificPackageName =
+       BINARY_DISTRIBUTION_PACKAGES[`${process.platform}-${process.arch}`]
+
+     try {
+       // Resolving will fail if the optionalDependency was not installed
+       return require.resolve(`${platformSpecificPackageName}/bin/${binaryName}`)
+     } catch (e) {
+       return require('path').join(__dirname, '..', binaryName)
+     }
+   }
+
+   // With `getBinaryPath()` could access the binary in you JavaScript code as follows
+   module.exports.runBinary = function (...args) {
+     require('child_process').execFileSync(getBinaryPath(), args, {
+       stdio: 'inherit',
+     })
+   }
+   ```
+
+4. _(You can skip this step if you do not need your binary to be accessible from the command line.)_
+   For the binary to be executable from the command line, we need to provide a script that invokes it.
+   Since the binary can be in two places, we create a wrapper script that locates the binary and invokes it.
+   We can reuse the `getBinaryPath()` function from before.
+
+   ```
+   #!/usr/bin/env node
+
+   require('child_process').execFileSync(getBinaryPath(), process.argv.slice(2), {
+     stdio: 'inherit',
+   })
+   ```
+
+   Save this script as `bin/cli` in `my-package`!
+   Additionally, add a `bin` field to your package.json:
+
+   ```jsonc
+   // package.json
+   {
+     "bin": {
+       "my-package": "bin/cli"
+     }
+   }
+   ```
+
+   This will cause your binary to be globally accessible when installed globally with a package manager, in addition to being directly invokable with a package manager.
+
+   ```sh
+   # Example: global installation
+   $ npm i -g my-package
+   $ my-package # will work from anywhere
+
+   # Example: invocation through package manager
+   $ npx my-package # will work from anywhere
+   ```
+
+5. Lastly, we must configure our main `package.json` to include the `optionalDependencies` and the `postinstall` script:
+
+   ```jsonc
+   {
+     "name": "my-package",
+     "version": "1.0.0",
+     "bin": {
+       "my-package": "bin/cli"
+     },
+     "scripts": {
+       "postinstall": "node ./install.js"
+     },
+     "optionalDependencies": {
+       "my-package-linux-x64": "1.0.0",
+       "my-package-linux-arm": "1.0.0",
+       "my-package-windows-x64": "1.0.0"
+     }
+   }
+   ```
+
+   To verify, the final package structure should look like the following:
+
+   ```
+   my-package
+   ├── package.json
+   ├── install.js
+   └── bin/
+       └── cli
+   ```
+
+Now we're done.
+Once `my-package` is published and installed, the binaries will be downloaded alongside, and you can either access them from your JS code with the `getBinaryPath()` function or you can directly invoke the binary executables from the command line.
+
```

Comment:
l/suggestion: Maybe add a "Wrapping Up"/conclusion paragraph to round off the post.

Reply:
Yup that is a good idea. I rephrased this part a bit! Thanks! :)